### PR TITLE
Rewrite README, move project from autotools to CMake

### PR DESCRIPTION
**Pull request by erysdren**

### CODE REVIEW

**Comment by erysdren** [comment]
Ah, I guess I should've also modified the workflows file too

File: README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+This port makes use of the Simple Direct Media Layer, or SDL, and an additional
+library named SDL_mixer.
+
+## Known Issues
+
+- Demos go out of sync.
+- No netplay support.
+- Won't build with MSVC.
+- Some crashes in the sound mixer.
```
Comment (fabiangreffrath) [inline README.md:18]: I think I have solved the sound mixer crashes here?

https://github.com/fabiangreffrath/rott/commit/45a9203d8160ccf0c045568fa4b5b93cba1d6191

File: README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+## Building
+
+### Linux
+
+Debian:
+```
+sudo apt-get install gcc build-essential cmake libsdl2-dev libsdl2-mixer-dev
+mkdir cmake-build && cd cmake-build
+cmake -DCMAKE_BUILD_TYPE=Release ..
+sudo make
```
Comment (fabiangreffrath) [inline README.md:29]: `sudo` is not required to call `make`.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## License
+
+GNU GPL v2
+
+Copyright (C) 1994-1995 Apogee Software, Ltd.
+
+Copyright (C) 2002-2015 Steven Fuller, Ryan C. Gordon, John Hall, Dan Olson
+
+Copyright (C) 2024 Fabian Greffrath
```
Comment (fabiangreffrath) [inline README.md:67]: Thank you, but that'd be `2006-2024` actually. 😉

File: README.md (hunk 1 above)
Comment (erysdren) [reply README.md:18]: I was getting a crash in my fork with the Excalibat at `FX_StopSound()` in `fx_mixer.c`. I couldn't figure out why, but somewhere along the way a value way higher than `MAX_CHANNELS` was getting passed into that function, causing an out of bounds memory access.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,47 @@
+
+#
```
[the remark this answers is not in the document]
Comment (erysdren) [reply CMakeLists.txt]: Done.

File: README.md (hunk 2 above)
Comment (erysdren) [reply README.md:29]: Done.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+mkdir cmake-build && cd cmake-build
+cmake -DCMAKE_BUILD_TYPE=Release ..
+sudo make
+```
+
+### Haiku
+
+```
+pkgman install gcc cmake libsdl2_devel sdl2_mixer_devel
+mkdir cmake-build && cd cmake-build
```
[the remark this answers is not in the document]
Comment (erysdren) [reply README.md:54]: Addressed in next commit.

File: README.md (hunk 3 above)
Comment (erysdren) [reply README.md:67]: Addressed in next commit.

File: README.md (hunk 1 above)
Comment (fabiangreffrath) [reply README.md:18]: Okay, that's a bug that I'd like to get reported, but probably not worth a generalized comment in the project's README. 😉

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -81,26 +81,33 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        if: runner.os == 'Windows'
+        with:
+          name: tarradino-${{ github.sha }}-win64
+          path: |
+            cmake-build/tarradino.exe
 
-      - name: Run cppcheck
-        shell: bash
-        run: |
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:104]: Why have you kicked out `cppcheck`?

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -81,26 +81,33 @@ jobs:
           CC: ${{ matrix.config.compiler }}
           CFLAGS: -Wall
         run: |
-          autoreconf -fiv
-          ./configure --enable-werror
+          cmake -E make_directory cmake-build
+          cmake -S . -B cmake-build
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:85]: We still need a way to compile with `-Werror` so that the build breaks if we introduce new warnings.

Comment (fabiangreffrath) [reply .github/workflows/main.yml:85]: Also, it would be nice to build both Shareware and Registered executable and upload both artifacts.

**Comment by fabiangreffrath** [comment]
There is still at least one `Makefile.am` file left in the source tree.

**Comment by fabiangreffrath** [comment]
Actually, the name is "Taradino", i.e. only one 'r'! 😮 

https://github.com/fabiangreffrath/rott/blob/ec15323a44b0526862320693abd66fb5b717c435/rott/rt_menu.c#L682

**Comment by erysdren** [comment]
I think the only thing missing from the CMakeLists.txt is the ability to change the executable name. Otherwise I think it's coming along nicely. CI seems to be failing from `-Werror`.

**Comment by fabiangreffrath** [comment]
> I think the only thing missing from the CMakeLists.txt is the ability to change the executable name. 

I'll see what I can do.

> CI seems to be failing from `-Werror`.

Yes, the shareware build hasn't been cleaned up yet.

**Comment by erysdren** [comment]
All looks good to me!

**Review by fabiangreffrath — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (fabiangreffrath) [reply README.md]: > I was getting a crash in my fork with the Excalibat at `FX_StopSound()` in `fx_mixer.c`. I couldn't figure out why, but somewhere along the way a value way higher than `MAX_CHANNELS` was getting passed into that function, causing an out of bounds memory access.

I think the problem is that many `SD_Play*()` functions return `0` to report failure, whereas `0` is indeed a valid channel number.

Comment (fabiangreffrath) [reply README.md]: > > I was getting a crash in my fork with the Excalibat at `FX_StopSound()` in `fx_mixer.c`. I couldn't figure out why, but somewhere along the way a value way higher than `MAX_CHANNELS` was getting passed into that function, causing an out of bounds memory access.
> 
> I think the problem is that many `SD_Play*()` functions return `0` to report failure, whereas `0` is indeed a valid channel number.

This should be solved by now.

**Pull request merged**